File: _bibliography/merge_bib.py
import os

BIB_DIR = "_bibliography"
OUTPUT_FILE = os.path.join(BIB_DIR, "references.bib")

# Collect all .bib files except references.bib
bib_files = [
    f for f in os.listdir(BIB_DIR)
    if f.endswith(".bib") and f != "references.bib"
]

merged = ""
for fname in bib_files:
    path = os.path.join(BIB_DIR, fname)
    with open(path, "r", encoding="utf-8") as f:
        content = f.read().strip()
        merged += f"% --- {fname} ---\n{content}\n\n"

# Write merged content
with open(OUTPUT_FILE, "w", encoding="utf-8") as out:
    out.write(merged)

print(f"✅ Merged {len(bib_files)} files into {OUTPUT_FILE}")

File: merge-bib.js
const fs = require("fs");
const path = require("path");

const bibDir = path.join(__dirname, "_bibliography");
const outputFile = path.join(bibDir, "references.bib");

// Get all .bib files in the folder except references.bib
const bibFiles = fs.readdirSync(bibDir)
  .filter(file => file.endsWith(".bib") && file !== "references.bib");

let merged = "";

bibFiles.forEach(file => {
  const filePath = path.join(bibDir, file);
  const content = fs.readFileSync(filePath, "utf8");
  merged += `% --- ${file} ---\n` + content.trim() + "\n\n";
});

fs.writeFileSync(outputFile, merged);
console.log(`✅ Merged ${bibFiles.length} files into references.bib`);
